docs/hugo/content/guide/authentication/credential-format.md

```diff
@@ -494,20 +516,121 @@ stringData:
 
+{{% /tab %}}
+{{% tab header="Namespace" %}}
+
+Deploy an `AzureIdentity`:
 ```bash
-helm repo add aso2 https://raw.githubusercontent.com/Azure/azure-service-operator/main/v2/charts
-helm repo update
+cat <<EOF | kubectl apply -f -
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentity
+metadata:
+  name: aso-identity
+  namespace: azureserviceoperator-system
+spec:
+  type: 0
+  resourceID: ${IDENTITY_RESOURCE_ID}
+  clientID: ${IDENTITY_CLIENT_ID}
+EOF
+```
 
-helm upgrade --install --devel aso2 aso2/azure-service-operator \
-     --create-namespace \
-     --namespace=azureserviceoperator-system \
-     --set azureSubscriptionID=$AZURE_SUBSCRIPTION_ID \
-     --set aadPodIdentity.enable=true \
-     --set aadPodIdentity.azureManagedIdentityResourceId=${IDENTITY_RESOURCE_ID} \
-     --set azureClientID=${IDENTITY_CLIENT_ID} \
-     --set crdPattern='resources.azure.com/*;containerservice.azure.com/*;keyvault.azure.com/*;managedidentity.azure.com/*;eventhub.azure.com/*'
+Deploy an `AzureIdentityBinding` to bind this identity to the Azure Service Operator manager pod:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentityBinding
+metadata:
+  name: aso-identity-binding
+  namespace: azureserviceoperator-system
+spec:
+  azureIdentity: aso-identity
+  selector: aso-manager-binding
+EOF
 ```
 
-See [CRD management]( {{< relref "crd-management" >}} ) for more details about `crdPattern`.
\ No newline at end of file
+Create the `aso-credential` secret in your namespace:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: aso-credential
+ namespace: my-namespace
+stringData:
+ AZURE_SUBSCRIPTION_ID: "$AZURE_SUBSCRIPTION_ID"
+ AZURE_TENANT_ID:       "$AZURE_TENANT_ID"
+ AZURE_CLIENT_ID:       "$IDENTITY_CLIENT_ID"
+ USE_POD_IDENTITY_AUTH: "true"
+EOF
+```
+
+{{% /tab %}}
+{{% tab header="Resource" %}}
+
+Deploy an `AzureIdentity`:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentity
+metadata:
+  name: aso-identity
+  namespace: azureserviceoperator-system
+spec:
+  type: 0
+  resourceID: ${IDENTITY_RESOURCE_ID}
+  clientID: ${IDENTITY_CLIENT_ID}
+EOF
+```
+
+Deploy an `AzureIdentityBinding` to bind this identity to the Azure Service Operator manager pod:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzureIdentityBinding
+metadata:
+  name: aso-identity-binding
+  namespace: azureserviceoperator-system
+spec:
+  azureIdentity: aso-identity
+  selector: aso-manager-binding
+EOF
+```
+
+Create a per-resource secret. We'll use `my-resource-secret`:
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: my-resource-secret
+ namespace: my-namespace
+stringData:
+ AZURE_SUBSCRIPTION_ID: "$AZURE_SUBSCRIPTION_ID"
+ AZURE_TENANT_ID:       "$AZURE_TENANT_ID"
+ AZURE_CLIENT_ID:       "$IDENTITY_CLIENT_ID"
+ USE_POD_IDENTITY_AUTH: "true"
```

Comment:
We already use a boolean for `USE_WORKLOAD_IDENTITY` for global secret and omit that for scoped credentials. Suggestions on using the same here for consistency and making forcing the Managed Identity to be a default in absence of ClientSecret?

Reply:
My thoughts on not doing the above : 
1. Users using Workload Identity would have to update their secrets to add a boolean
2. Pod Identity is deprecated, so did not want to use it as default.

Reply:
The problem with having multiple boolean configurations like this is that we end up with oddities - what should happen if both `USER_POD_IDENTITY_AUTH` and `USE_WORKLOAD_IDENTITY` are **true** at the same time?

I'd like us to consider using a single new configuration setting and deprecating `USE_WORKLOAD_IDENTITY`. 

What if we introduced `IDENTITY_AUTH` with permitted values of `pod`, `workload` or `scoped`, with `scoped` as the default if not set. 
For back compat, `USE_WORKLOAD_IDENTITY: "true"` would change the default to `workload`.

Reply:
> The problem with having multiple boolean configurations like this is that we end up with oddities - what should happen if both USER_POD_IDENTITY_AUTH and USE_WORKLOAD_IDENTITY are true at the same time?

AFAIK we don't actually support `USE_POD_IDENTITY_AUTH` and `USE_WORKLOAD_IDENTITY_AUTH` on the same secrets.
`USE_POD_IDENTITY_AUTH` is allowed only in the scoped credential secret (namespace or per-resource). It's not allowed at the global secret level from what I can tell.

`USE_WORKLOAD_IDENTITY` is only at the global secret level, it's not supported at the per-resource/per-namespace secret (@super-harsh correct me if I am wrong here). Instead, it's always inferred if the user omits `AZURE_CLIENT_SECRET` that they want workload identity.

So there's no _technical_ ambiguity here even if there may be some user-ambiguity. It would be better if things were exactly the same between the different secret formats probably, but I do think how we got here is somewhat reasonable because:

1. The global secret used the "if no `AZURE_CLIENT_SECRET`, assume AAD Pod Identity" approach, since it previously supported AAD Pod Identity. When we added workload identity support we needed a way to differentiate between the existing supported AAD Pod Identity and Workload Identity. We possibly should've added an enum instead of a boolean here, but the choice of "evolve without breaking" seems right to me.
2. The per-namespace and per-resource secrets didn't yet exist, so didn't support AAD Pod Identity, so didn't need a boolean until now. Now, we want to add support for AAD Pod Identity in a nonbrekaing way, so we need to differentiate between workload identity (already supported, chosen by omitting the `AZURE_CLIENT_SECRET`) and this new (legacy) mode of AAD Pod Identity.

I think the direction we want to head (eventually) is that AAD Pod Identity is fully deprecated and we can drop all these flags and just have the "if client_secret omitted, workload identity" - though maybe we want an enum just to allow future evolution. I don't see how we get:
1. Global and per-namespace/per-resource secrets are consistently shaped (support the same fields/flags).
2. No breaking changes from what we have today.

Without doing something like what @super-harsh has done here (though agree maybe boolean worse than enum)